subsys/ipc/rpmsg_multi_instance/rpmsg_multi_instance.c

```diff
@@ -100,11 +97,27 @@ static void ipm_callback(const struct device *dev, void *context, uint32_t id, v
 	k_work_submit_to_queue(&ctx->ipm_work_q, &ctx->ipm_work);
 }
 
+static inline unsigned int get_instance(struct rpmsg_mi_ctx_shm_cfg *shm)
+{
+	unsigned int this_instance;
+	struct k_spinlock shm_lock;
+	k_spinlock_key_t key;
+
+	key = k_spin_lock(&shm_lock);
+	this_instance = shm->instance;
+	shm->instance++;
+	k_spin_unlock(&shm_lock, key);
```

Comment:
I see that this is still unresolved, see: https://github.com/zephyrproject-rtos/zephyr/pull/37361#discussion_r685187779

Reply:
Uhm, I'm probably missing something but I implemented your suggestion at that comment.

Reply:
It's still in the PR, so maybe some merging problem? Please make sure that you pushed right commit to this PR.

Reply:
I'm still lost, sorry. It's in the PR because you suggested that change in the https://github.com/zephyrproject-rtos/zephyr/pull/37361#discussion_r685187779 so I implemented that. The only difference with what you suggested is that now the code is inside a `get_instance()` inline function.

Reply:
Sorry, my mistake. You responded with *ACK* after my suggestion with atomics, so I was thinking that you will implement it. I see that you implemented my first suggestion, which is also ok, so I will accept the PR.